### DISCUSSION TITLE
📸 Introducing Publii Gallery Plugins!

### DISCUSSION THREAD

**Discussion by bobmitro** [GetPublii/Publii#1869] [General]
Hey there!

We’ve launched two new gallery plugins to make managing and displaying images on your Publii site easier and more flexible! 🖼️
Learn more about the Core Lightbox Gallery and Simple Lightbox Gallery and how they can enhance your site's functionality.

👉 Check out the full announcement: [Publii Gallery Plugins](https://getpublii.com/blog/publii-gallery-plugins.html)

**Comment by jpbraun** [comment]
Hi Bob,
that is great. Thank you very much!

**Comment by henkvantijen** [comment]
Great news...  Color me stupid, but I installed the two gallery plugins, activated those with the radio buttons,  upgraded to Simple 3.1, app is on 0.64.2 ... But as far as I can see the gallery's I can insert are still the old ones? How do I install a new gallery on a page / post ?

At Custom Theme Settings it looks like I still must install the gallery ?

<img width="676" alt="image" src="https://github.com/user-attachments/assets/0e5c6771-e459-4918-a496-2f841afde0ac" />

**Reply by bobmitro** [reply]
> But as far as I can see the gallery's I can insert are still the old ones? 

I'm not sure I understand you well. If you installed the new theme version and didn’t override the `head.hbs` and `footer.hbs` files, it’s impossible to see the old gallery. The only thing that comes to mind is the cache.

And don’t enable two galleries, just one, because two won’t work correctly.

**Reply by henkvantijen** [reply]
Ahh... Thx.  _Of course_ I enabled both  , I assumed you would be able to choose between the two in the content of page or post 😬. But I understand it now.  
Disabled 1 and it works! Thx. 🙏🏽

**Comment by followmyrandomness** [comment]
Previously, I have customised my website to also have a gallery functionality on the blog/tag pages, not just inside the posts by enabling <link rel="stylesheet" href="{{css "photoswipe.css" }}"> in head.hbs regardless of whether this was post/page, index, or whatever.

Is there any way of achieving the same with the new gallery plugin system?

Viewing the  html source, I can see the "link rel" to the chosen gallery's css being placed roughly before where the  {{{@headCustomCode}}} would be in the header.hbs, but can't figure whether there is a way to manually link to the galleryy's css in my header.hbs

**Comment by bobmitro** [comment]
The plugin now works differently, and `hasGallery` flag only works in the context of posts and pages, if I remember correctly. However, it can be easily modified by updating the plugin code:

In the `main.js` file (plugin), find the line with the following code (it appears in two places):

`if (!context.post?.hasGallery && !context.page?.hasGallery) return '';
`

And replace it with:

`if (!context.post?.hasGallery && !context.page?.hasGallery && !context.tag) return '';
`

The plugin should now work on tag pages within the tag context.

**Reply by followmyrandomness** [reply]
Thanks, I'll give it a try.  I guess I can potentially remove that "if" clause altogether if I wanted to have gallery functionality anywhere on the website?

The only potential drawback in this method (I think, I might be wrong) is that it has to be done again every time the plugin updated. Do you think you could perhaps considering implementing this into the plugin settings? 

Thank you

**Reply by bobmitro** [reply]
> Do you think you could perhaps considering implementing this into the plugin settings?

I don’t think so. I didn’t choose this approach because the gallery is available only in posts and pages. Additionally, thanks to the `hasGallery` flag, we can detect whether a gallery exists, allowing the plugin to load scripts only on selected pages. Without this, the scripts would be loaded on all pages, which is far from optimal. 

In general, custom work requires custom solutions. :)

**Reply by followmyrandomness** [reply]
> In general, custom work requires custom solutions. :)

Fair enough, can't argue with that 😁

**Reply by emanuelefavaro** [reply]
Hi there, 
I come back to this topic because I need to add the gallery scripts and styles to an home page but I can't find a way to do that. 
I'm using the Square theme and trying to show a post which consists of a single lightbox gallery in the feed on the right column. 
The gallery works fine if I open the post, but on the homepage only if I manually add the necessary scripts and styles to the code.
I can't find a way to make the thing work dynamically, modyfing `main.js`, as suggested above (I've tried adding some clauses to the `if` statement but with no luck).
Am I missing something? Thanks!
